src/dataset.rs

```diff
@@ -67,42 +119,67 @@ impl Dataset {
         }
     }
 
-    pub fn size(&self) -> (usize, usize) {
-        let size_x = unsafe { gdal_sys::GDALGetRasterXSize(self.c_dataset) } as usize;
-        let size_y = unsafe { gdal_sys::GDALGetRasterYSize(self.c_dataset) } as usize;
-        (size_x, size_y)
+    fn child_layer(&self, c_layer: OGRLayerH) -> Layer {
+        let layer = unsafe { Layer::_with_c_layer(c_layer, self) };
+        layer
     }
 
-    /// Get block size from a 'Dataset'.
-    /// # Arguments
-    /// * band_index - the band_index
-    /*
-    pub fn size_block(&self, band_index: isize) -> (usize, usize) {
-        let band = self.rasterband(band_index)?;
-        band.size_block()
+    pub fn layer_count(&self) -> isize {
+        (unsafe { gdal_sys::OGR_DS_GetLayerCount(self.c_dataset) }) as isize
     }
-    */
 
-    pub fn driver(&self) -> Driver {
-        unsafe {
-            let c_driver = gdal_sys::GDALGetDatasetDriver(self.c_dataset);
-            Driver::_with_c_ptr(c_driver)
+    pub fn layer(&mut self, idx: isize) -> Result<Layer> {
+        let c_layer = unsafe { gdal_sys::OGR_DS_GetLayer(self.c_dataset, idx as c_int) };
+        if c_layer.is_null() {
+            Err(_last_null_pointer_err("OGR_DS_GetLayer"))?;
         }
+        Ok(self.child_layer(c_layer))
     }
 
-    pub fn count(&self) -> isize {
+    pub fn layer_by_name(&mut self, name: &str) -> Result<Layer> {
+        let c_name = CString::new(name)?;
+        let c_layer = unsafe { gdal_sys::OGR_DS_GetLayerByName(self.c_dataset(), c_name.as_ptr()) };
+        if c_layer.is_null() {
+            Err(_last_null_pointer_err("OGR_DS_GetLayerByName"))?;
+        }
+        Ok(self.child_layer(c_layer))
+    }
+
+    pub fn raster_count(&self) -> isize {
```

Comment:
What happens if the dataset is no raster? Same as `layer_count` if the dataset is a raster.

Reply:
Since GDAL 2 there is only one Dataset: https://gdal.org/doxygen/classGDALDataset.html . The count will be 0 in the respective case.